spec/unit/matrixrtc/MatrixRTCSession.spec.ts

```diff
@@ -477,23 +477,25 @@ describe("MatrixRTCSession", () => {
                         return Promise.reject(error);
                     });
                 });
+                // needed to join so that myMembershipManager gets created
+                sess!.joinRoomSession([activeFocusConfig], activeFocus, {
+                    membershipServerSideExpiryTimeout: 9000,
+                });
 
                 // needed to advance the mock timers properly
+                // depends on myMembershipManager being created
                 const scheduledDelayDisconnection = new Promise<void>((resolve) => {
-                    const originalFn: () => void = (sess as any).scheduleDelayDisconnection;
-                    (sess as any).scheduleDelayDisconnection = jest.fn(() => {
-                        originalFn.call(sess);
+                    const myMemManager = (sess as any).myMembershipManager;
+                    const originalFn: () => void = myMemManager.scheduleDelayDisconnection;
+                    myMemManager.scheduleDelayDisconnection = jest.fn(() => {
+                        originalFn.call(myMemManager);
                         resolve();
                     });
                 });
 
-                sess!.joinRoomSession([activeFocusConfig], activeFocus, {
-                    membershipServerSideExpiryTimeout: 9000,
-                });
-
-                expect(sess).toHaveProperty("membershipServerSideExpiryTimeout", 9000);
+                expect((sess as any).myMembershipManager).toHaveProperty("membershipServerSideExpiryTimeout", 9000);
```

Comment:
As long as you're working on creating a testable, public interface for the session manager we should ideally avoid inspecting private fields like this and test the behavior they govern instead

Reply:
This is indeed very easy and much better.